web3.js/src/instruction.ts

```diff
@@ -3,21 +3,28 @@ import * as BufferLayout from '@solana/buffer-layout';
 
 import * as Layout from './layout';
 
+export interface IInstructionInputData {
+  readonly instruction: number;
+}
+
 /**
  * @internal
  */
-export type InstructionType = {
+export type InstructionType<TInputData extends IInstructionInputData> = {
```

Comment:
What we're saying here is that every input buffer contains at least an instruction number.